### DISCUSSION TITLE
Tiffs with embedded thumbnails, 16 and 8 bit?

### DISCUSSION THREAD

**Discussion by iprintJay** [ImageMagick/ImageMagick#6637] [Help]
### ImageMagick version

7.1.1-14

### Operating system, version and so on

Windows 10

### Description

I am having issues processing certain tiff files, when I open them in a program like Krita they show a second thumbnail layer in the upper left corner. It seems like adding a [0] to my IM command should take care of this but some files are stubborn. The command `magick identify -format %e:%m:%n:%[bit-depth]:%[colorspace]` will result in this:

```
identify: Invalid TIFF directory; tags are not sorted in ascending order. `TIFFReadDirectoryCheckOrder' @ warning/tiff.c/TIFFWarnings/1044.
tiff:TIFF:2:16:sRGBtiff:TIFF:2:8:sRGB
```

and if I process the image it is really blown out with bright pastel colors.  A verbose readout shows some interesting things, such as this on the first layer (Krita thinks the image is straight up 8-bit): 
```
Depth: 8/16-bit
  Channel depth:
    Red: 16-bit
    Green: 16-bit
    Blue: 16-bit
    Alpha: 8-bit
```
And the thumbnail layer as all 8-bit, no alpha, with this little bit: `tiff:subfiletype: REDUCEDIMAGE`

Is there any way to process these tiffs properly? Also the odd processing happens in older versions, 8 and 16 bit versions. Thanks

**Comment by snibgo** [comment]
Warnings don't prevent further processing, so don't worry about that.

The only problem you seem to have is "it is really blown out with bright pastel colors".

You mention embedded thumbnails, but don't say what whether you want to extract them, or what.

Advice is difficult. If you post a sample file, perhaps we can help.
